### PR TITLE
qbs: Update to 1.9.0 (maintainer)

### PR DESCRIPTION
**Pull request by jakepetroules**

### CODE REVIEW

File: devel/qbs/Portfile [hunk 1]
```diff
@@ -30,10 +30,6 @@ configure.post_args QBS_INSTALL_PREFIX=${prefix} \
                     CONFIG+=qbs_enable_project_file_updates \
                     qbs.pro
 
-# as of qbs 1.8.1, tools/version.h needs to be installed
-# see https://github.com/qt-labs/qbs/commit/7ccc08f33153cd3093081567ac653e371759d69a#diff-5958438bc65bf7bd9145fcfed6cde03d
-patchfiles-append   patch-version_h.diff
```
Comment (l2dy) [inline devel/qbs/Portfile]: Please remove the patchfile too.

File: devel/qbs/Portfile [hunk 2]
```diff
@@ -6,8 +6,8 @@ PortGroup           cxx11 1.1
 
 name                qbs
 
-version             1.8.1
-revision            2
+version             1.9.0
+revision            1
```
Comment (l2dy) [inline devel/qbs/Portfile]: `revision` starts from 0, you should set it to 0 or remove this line.

**Comment by l2dy** [comment]
Please add your GitHub username to the maintainers line in the Portfile as [described in the guide](https://guide.macports.org/#reference.keywords.maintainers).

File: devel/qbs/Portfile (hunk 2 above)
Comment (jakepetroules) [reply devel/qbs/Portfile]: Done

File: devel/qbs/Portfile (hunk 1 above)
Comment (jakepetroules) [reply devel/qbs/Portfile]: Done

**Comment by jakepetroules** [comment]
GitHub username added.

**Comment by jakepetroules** [comment]
Travis doesn't give any info why the build failed...

**Comment by l2dy** [comment]
There's a link to build logs on Travis: https://paste.macports.org/ef951fc75c0d
```
--->  Configuring qbs
DEBUG: Preferred compilers: clang macports-clang-3.9 macports-clang-3.8 macports-clang-3.7
DEBUG: Using compiler 'Xcode Clang'
DEBUG: Executing proc-pre-org.macports.configure-configure-0
DEBUG: qt5 PortGroup: Qt is provided by qt5
DEBUG: Executing proc-pre-org.macports.configure-configure-1
DEBUG: Active variants check for source-type install considers depends_fetch depends_extract depends_lib depends_build depends_run: qt5-qtbase qt5-qtscript pkgconfig
DEBUG: Executing proc-pre-org.macports.configure-configure-2
Error: Failed to configure qbs: couldn't open "/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_devel_qbs/qbs/work/qbs-src-1.9.0/.qmake.cache": no such file or directory
DEBUG: Error code: NONE
DEBUG: Backtrace: couldn't open "/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_devel_qbs/qbs/work/qbs-src-1.9.0/.qmake.cache": no such file or directory
    while executing
"$pre $targetname"
Error: See /opt/local/var/macports/logs/_Users_travis_build_macports_macports-ports_devel_qbs/qbs/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets to report a bug.
Error: Processing of port qbs failed
```

**Comment by l2dy** [comment]
Do you think this is a Travis-specific issue?

**Comment by jakepetroules** [comment]
Could be something in the Travis environment, I can't see why that would fail like that. @ossilator, any idea?

**Comment by l2dy** [comment]
Build failed on [our Bulidbot](https://build.macports.org/builders/ports-10.12_x86_64-builder/builds/38053/steps/install-port/logs/stdio) too, so this is not specific to Travis. Please fix this soon.

**Comment by l2dy** [comment]
Sorry for merging this too soon, I thought it was something with Travis environment.

**Comment by jakepetroules** [comment]
Not sure what's causing this. Will forward to a colleague.

**Comment by jakepetroules** [comment]
It succeeded on 10.10 and 10.11 but failed on 10.12, right?

**Comment by l2dy** [comment]
No, it failed on all macOS versions.

**Comment by ossilator** [comment]
i don't have the slightest clue.
was there a successful build since the last time qtbase/qmake was updated?
how does that system even know that it's supposed to call qmake? maybe the surrounding code makes some bogus assumptions? at first sight, the error message doesn't appear to come from qmake or qbs' project files.

**Comment by jakepetroules** [comment]
I don't see how we've done anything in qbs that could have caused this; I wonder if it's something wrong with the packaging of qt5 in MacPorts? Have we noticed any similar qmake issue elsewhere and/or has MacPorts seen any other packages depending on qt5 fail with the same issue recently?

**Comment by l2dy** [comment]
See https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/qmake5-1.0.tcl#L94.

**Comment by l2dy** [comment]
The port uses the qmake5 PortGroup which determines how to call qmake. I have no clue how it failed though. Should I revert the merged commit so users don't get a broken port?

**Comment by jakepetroules** [comment]
This seems like an issue in qmake5-1.0.tcl. Is it assuming .qmake.cache already exists before opening it? We've never had such a file in the qbs repository.

**Comment by l2dy** [comment]
The old Portfile seems to work (#782 timed out while this PR errored on Travis) and both uses the same PortGroup.

According to https://www.tcl.tk/man/tcl8.4/TclCmd/open.htm, it doesn't seem to assume that .qmake.cache already exists.

> w
> Open the file for writing only. Truncate it if it exists. If it doesn't exist, create a new file.

Should I merge the revert PR #782?

**Comment by l2dy** [comment]
Fixed in f53cb73c204c336d187902eaaeca53e34ec3e727.

**Pull request merged**